homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,230 @@
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(hass, session, hivedevice)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hass, Session, HiveDevice):
+        """Initialize the Climate device."""
+        self.node_id = HiveDevice["Hive_NodeID"]
+        self.node_name = HiveDevice["Hive_NodeName"]
+        self.device_type = HiveDevice["HA_DeviceType"]
+        self.hass = hass
+        self.session = Session
+
+        if self.device_type == "Heating":
+            set_entity_id = "Hive_Heating"
+            self.session.heating = self.session.core.Heating()
+            self.modes = [STATE_AUTO, STATE_HEAT, STATE_OFF]
+        elif self.device_type == "HotWater":
+            set_entity_id = "Hive_HotWater"
+            self.session.hotwater = self.session.core.Hotwater()
+            self.modes = [STATE_AUTO, STATE_ON, STATE_OFF]
+        if self.node_name is not None:
+            set_entity_id = set_entity_id + "_" \
+                            + self.node_name.replace(" ", "_")
+        self.entity_id = ENTITY_ID_FORMAT.format(set_entity_id.lower())
+
+        self.session.entities.append(self)
+
+    def handle_update(self, updatesource):
+        """Handle the new update request."""
+        if self.device_type + "." + self.node_id not in updatesource:
+            self.schedule_update_ha_state()
+
+    @property
+    def name(self):
+        """Return the name of the Climate device."""
+        friendly_name = "Climate Device"
+        if self.device_type == "Heating":
+            friendly_name = "Heating"
+            if self.node_name is not None:
+                friendly_name = self.node_name + " " + friendly_name
+
+        elif self.device_type == "HotWater":
+            friendly_name = "Hot Water"
+
+        return friendly_name
+
+    @property
+    def force_update(self):
+        """Return True if state updates should be forced."""
+        return False
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement which this device uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self.device_type == "Heating":
+            nodeid = self.node_id
+            curtempret = self.session.heating.current_temperature(nodeid)
+
+            if curtempret != -1000:

```

Comment:
This seems to be some magical constant? Should it indicate "not set" or something? Perhaps setting it to `None` would be better?

Reply:
it is so that it can compare current temperature for minimum and maximum with values that should never be reached and then return the relevant value at the end